appcenter/android/src/main/java/com/microsoft/appcenter/reactnative/appcenter/AppCenterReactNativeListener.java

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.appcenter.reactnative.appcenter;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.modules.core.DeviceEventManagerModule;
+import com.microsoft.appcenter.AuthTokenCallback;
+import com.microsoft.appcenter.AuthTokenListener;
+
+public class AppCenterReactNativeListener implements AuthTokenListener {
+
+    private static final String ON_ACQUIRE_AUTH_TOKEN_EVENT = "OnAcquireAuthTokenEvent";
+
+    private ReactApplicationContext mReactApplicationContext;
+
+    private AuthTokenCallback mCallback;
+
+    private boolean mAuthTokenEventSent = false;
+
+    private boolean mHasPendingEvent = false;
+
+    @SuppressWarnings("WeakerAccess")
+    public final void setReactApplicationContext(ReactApplicationContext reactApplicationContext) {
+        mReactApplicationContext = reactApplicationContext;
+    }
+
+    @Override
+    public void acquireAuthToken(AuthTokenCallback callback) {
+        ReactNativeUtils.logInfo("Auth token requested from native code.");
+        sendEvent(ON_ACQUIRE_AUTH_TOKEN_EVENT);
+        mCallback = callback;
+    }
+
+    private void sendEvent(String eventType) {
+        if (mReactApplicationContext != null) {
+            if (mHasPendingEvent) {
+                mReactApplicationContext
+                        .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                        .emit(eventType, null);
+                mAuthTokenEventSent = true;
```

Comment:
After emitting, just switch mHasPendingEvent to false and then you don't need the second boolean.